test/node/double_tests.js

```diff
@@ -38,26 +38,82 @@ describe('BSON Double Precision', function () {
-        { input: new Double(3.4), output: { $numberDouble: '3.4' } },
-        { input: new Double(Number.EPSILON), output: { $numberDouble: '2.220446049250313e-16' } },
-        { input: new Double(12345e7), output: { $numberDouble: '123450000000.0' } },
-        { input: new Double(12345e-1), output: { $numberDouble: '1234.5' } },
-        { input: new Double(-12345e-1), output: { $numberDouble: '-1234.5' } },
-        { input: new Double(Infinity), output: { $numberDouble: 'Infinity' } },
-        { input: new Double(-Infinity), output: { $numberDouble: '-Infinity' } },
-        { input: new Double(NaN), output: { $numberDouble: 'NaN' } }
+        { input: 0, output: { $numberDouble: '0.0' } },
+        { input: -0, output: { $numberDouble: '-0.0' } },
+        { input: '-0.0', output: { $numberDouble: '-0.0' } },
+        { input: 3, output: { $numberDouble: '3.0' } },
+        { input: -3, output: { $numberDouble: '-3.0' } },
+        { input: 3.4, output: { $numberDouble: '3.4' } },
+        { input: Number.EPSILON, output: { $numberDouble: '2.220446049250313e-16' } },
+        { input: 12345e7, output: { $numberDouble: '123450000000.0' } },
+        { input: 12345e-1, output: { $numberDouble: '1234.5' } },
+        { input: -12345e-1, output: { $numberDouble: '-1234.5' } },
+        { input: Infinity, output: { $numberDouble: 'Infinity' } },
+        { input: -Infinity, output: { $numberDouble: '-Infinity' } },
+        { input: NaN, output: { $numberDouble: 'NaN' } },
+        {
+          input: Number.MAX_VALUE,
+          output: { $numberDouble: '1.7976931348623157e+308' }
+        },
+        { input: Number.MIN_VALUE, output: { $numberDouble: '5e-324' } },
+        {
+          input: -Number.MAX_VALUE,
+          output: { $numberDouble: '-1.7976931348623157e+308' }
+        },
+        { input: -Number.MIN_VALUE, output: { $numberDouble: '-5e-324' } },
+        // Reference: https://docs.oracle.com/cd/E19957-01/806-3568/ncg_math.html
+        {
+          // min positive normal number
+          input: '2.2250738585072014e-308',
+          output: { $numberDouble: '2.2250738585072014e-308' }
+        },
+        {
+          // max subnormal number (NOTE: JS does not output same input string, but numeric values are equal)
+          input: '2.225073858507201e-308',
+          output: { $numberDouble: '2.225073858507201e-308' }
+        },
+        {
+          // min positive subnormal number (NOTE: JS does not output same input string, but numeric values are equal)
+          input: '4.9406564584124654e-324',
+          output: { $numberDouble: '5e-324' }
+        },
+        {
+          // https://262.ecma-international.org/13.0/#sec-number.prototype.tofixed
+          // Note: calling toString on this integer returns 1000000000000000100, so toFixed is more precise
+          input: '1000000000000000128',
+          output: { $numberDouble: '1000000000000000128.0' }
+        }
```

Comment:
while this is interesting, I'm not sure this provides any additional value.  We already have tests that show that both positive and negative integer numbers serialize to numbers with a decimal.

Reply:
This is an integer that's beyond the precision that an 8 byte float can hold (hence why toString loses the 28), it confirms that we still output a string that preserves it if that was the input. (Basically, asserts that toFixed is being used for ints).